airflow/www/extensions/init_views.py

```diff
@@ -171,6 +185,7 @@ def _handle_api_error(ex):
     api_bp = connexion_app.add_api(
         specification='v1.yaml', base_path=base_path, validate_responses=True, strict_validation=True
     ).blueprint
+    app.after_request(set_cors_headers_on_response)
```

Comment:
```suggestion
    api_bp.after_request(set_cors_headers_on_response)
```

Reply:
Connexion is doing something funny -- this doesn't work (the after request is never called!)

Reply:
Ah, the problem is this blueprint is already registered when it's returned, so using `bp.after_request` doesn't get added to the flask app's callback list.

Reply:
```suggestion
    # Like "api_bp.after_request", but the BP is already registered, so we have to
    # register it in the app directly.
    app.after_request_funcs.setdefault(api_bp.name, []).append(set_cors_headers_on_response)

```